### PR TITLE
[Refactor] Simplify home screen features and cards update logic

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
This PR aims at:
 * Moving more business logic (i.e. MapConfig) from Fragments to ViewModels
 * Replacing chaining of flowables/flow in HomeScreenMapContainerViewModel to more synchronous code
 * Removing duplicate map update calls

@gino-m  PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt [hunk 1]
```diff
@@ -89,61 +87,76 @@ internal constructor(
    * List of [LocationOfInterest] for the active survey that are present within the map bounds and
    * zoom level is clustering threshold or higher.
    */
-  val loisWithinMapBoundsAtVisibleZoomLevel: LiveData<List<LocationOfInterest>>
+  private val _loisWithinMapBoundsAtVisibleZoomLevel: MutableStateFlow<List<LocationOfInterest>> =
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt:90]: s/...WithinMapBoundsAtVisibleZoomLevel/...InViewport/ here and throughout?

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt [hunk 2]
```diff
@@ -89,61 +87,76 @@ internal constructor(
...
+    } else {
+      // LOIs that are persisted to the local and remote dbs.
+      _mapLoiFeatures.emitAll(locationOfInterestRepository.findLocationsOfInterestFeatures(survey))
+    }
+  }
+
+  private suspend fun updateMapLois(survey: Survey?, cameraPosition: CameraPosition?) {
+    val bounds = cameraPosition?.bounds
+    val isZoomLevelLessThanThreshold = cameraPosition.isLessThanThreshold(CLUSTERING_ZOOM_THRESHOLD)
+    if (bounds == null || survey == null || isZoomLevelLessThanThreshold) {
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt:135]: Not sure the added indirection of the helper here adds clarity. 

As an alternative, you could get rid of `val isZoomLevelLessThanThreshold`, rename the helper `isBelowClusteringZoomThreshold` and move the constant into the helper. Then you'd end up with:

```kt
   val bounds = cameraPosition?.bounds
    if (bounds == null || survey == null || cameraPosition.isBelowZoomThreshold()) {
   ...
    }
```

Or similar.

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt [hunk 3]
```diff
@@ -89,61 +87,76 @@ internal constructor(
...
+    } else {
+      _loisWithinMapBoundsAtVisibleZoomLevel.emitAll(
+        locationOfInterestRepository.getWithinBoundsOnceAndStream(survey, bounds).asFlow()
+      )
+    }
+  }
+
+  private fun updateSuggestLoiJobs(survey: Survey?, cameraPosition: CameraPosition?) {
+    val isZoomLevelLessThanThreshold = cameraPosition.isLessThanThreshold(CLUSTERING_ZOOM_THRESHOLD)
+    if (survey == null || isZoomLevelLessThanThreshold) {
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt:146]: Same here.

**Review by gino-m — CHANGES_REQUESTED** [review]
Nice work! LG, just a few nits. See also suggestions for future work in #1853.

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt (hunk 1 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt]: Done

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt (hunk 2 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt]: Done

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt (hunk 3 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt]: Done

**Review by gino-m — APPROVED** [review]
LGTM! 🚀

**Pull request merged**